service/task.go

```diff
@@ -507,140 +493,122 @@ func getTaskDependencies(t *task.Task) ([]uiDep, string, error) {
+		if err != nil {
+			http.Error(w, "invalid execution", http.StatusBadRequest)
+			return
+		}
+
+		if tsk.Execution != execution {
+			tsk, err = task.FindOneIdAndExecution(tsk.Id, execution)
+			if err != nil {
+				uis.LoggedError(w, r, http.StatusInternalServerError, err)
+				return
+			}
+			if tsk == nil {
+				http.Error(w, fmt.Sprintf("task '%s' with execution '%d' not found", projCtx.Task.Id, execution), http.StatusNotFound)
+				return
+			}
+		}
 	}
+
 	logType := r.FormValue("type")
 	if logType == "EV" {
-		var loggedEvents []event.EventLogEntry
-		loggedEvents, err = event.Find(event.MostRecentTaskEvents(projCtx.Task.Id, DefaultLogMessages))
+		loggedEvents, err := event.Find(event.MostRecentTaskEvents(projCtx.Task.Id, DefaultLogMessages))
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+			uis.LoggedError(w, r, http.StatusInternalServerError, err)
 			return
 		}
+
 		gimlet.WriteJSON(w, loggedEvents)
 		return
 	}
-	ctx := r.Context()
 
-	// check buildlogger logs first
-	opts := apimodels.GetBuildloggerLogsOptions{
-		BaseURL:       uis.Settings.Cedar.BaseURL,
-		TaskID:        projCtx.Task.Id,
-		Execution:     utility.ToIntPtr(execution),
-		PrintPriority: true,
-		Tail:          DefaultLogMessages,
-		LogType:       logType,
-	}
-	var logReader io.ReadCloser
-	logReader, err = apimodels.GetBuildloggerLogs(ctx, opts)
-	if err == nil {
-		defer func() {
-			grip.Warning(message.WrapError(logReader.Close(), message.Fields{
-				"task_id": projCtx.Task.Id,
-				"message": "failed to close buildlogger log ReadCloser",
-			}))
-		}()
-		gimlet.WriteJSON(w, apimodels.ReadBuildloggerToSlice(ctx, projCtx.Task.Id, logReader))
+	it, err := tsk.GetTaskLogs(r.Context(), uis.env, taskoutput.TaskLogGetOptions{
+		LogType: getTaskLogTypeMapping(logType),
+		TailN:   DefaultLogMessages,
+	})
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, err)
 		return
 	}
-	grip.Warning(message.WrapError(err, message.Fields{
-		"task_id": projCtx.Task.Id,
-		"message": "problem getting buildlogger logs",
-	}))
 
-	taskLogs, err := getTaskLogs(projCtx.Task.Id, execution, DefaultLogMessages, logType)
+	lines, err := apimodels.ReadLogToSlice(it)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		uis.LoggedError(w, r, http.StatusInternalServerError, err)
 		return
 	}
-	wrapper := &taskLogsWrapper{LogMessages: taskLogs}
-	gimlet.WriteJSON(w, wrapper)
+
+	gimlet.WriteJSON(w, struct {
+		LogMessages []*apimodels.LogMessage
+	}{LogMessages: lines})
 }
 
 func (uis *UIServer) taskLogRaw(w http.ResponseWriter, r *http.Request) {
 	projCtx := MustHaveProjectContext(r)
 	if projCtx.Task == nil {
-		http.Error(w, "Not found", http.StatusNotFound)
+		http.Error(w, "task not found", http.StatusNotFound)
 		return
 	}
 
-	raw := (r.FormValue("text") == "true") || (r.Header.Get("Content-Type") == "text/plain")
-	printPriority := r.FormValue("priority") == "true" || !raw
```

Comment:
Looks like this is missing priority information if the `priority=true` is set.

Before:
![image](https://github.com/evergreen-ci/evergreen/assets/4605522/88e712bf-5190-45ce-bfe8-453593ee1fe0)

After:
![image](https://github.com/evergreen-ci/evergreen/assets/4605522/7e718ad0-2d6f-461a-81c5-bca9c12c883e)

Reply:
Parsley passes the following query params to request task logs.
`?priority=true&text=true&type=T`
It looks like priority might be getting handled correctly.

Reply:
Fixed!
![Screenshot 2023-11-01 at 11 48 06 AM](https://github.com/evergreen-ci/evergreen/assets/16562432/4d6f068a-99a5-44d0-baf6-f472d7c15c12)